### ISSUE TITLE
quantized models in wasm throws ERROR not all tensors loaded from model file 

### ISSUE THREAD

**Issue by jozefchutka** [ggerganov/whisper.cpp#799]
The demo on https://whisper.ggerganov.com/ works ok with quantized models (i.e. base.en) .

However, when building wasm from the latest master:

```sh
cd whisper.cpp
mkdir build-em && cd build-em
emcmake cmake ..
make -j
```

the very same quantized model fails to initialize:

```
const modelBuffer = await (await fetch("ggml-model-whisper-base.en-q4_0.bin")).arrayBuffer();
const model = new Uint8Array(modelBuffer);
Module.FS_createDataFile("/", "whisper.bin", model, true, true);
Module.init("whisper.bin");
```

throwing:

```
whisper_init_from_file_no_state: loading model from 'whisper.bin'
main.js:1 whisper_model_load: loading model
main.js:1 whisper_model_load: n_vocab       = 51864
main.js:1 whisper_model_load: n_audio_ctx   = 1500
main.js:1 whisper_model_load: n_audio_state = 512
main.js:1 whisper_model_load: n_audio_head  = 8
main.js:1 whisper_model_load: n_audio_layer = 6
main.js:1 whisper_model_load: n_text_ctx    = 448
main.js:1 whisper_model_load: n_text_state  = 512
main.js:1 whisper_model_load: n_text_head   = 8
main.js:1 whisper_model_load: n_text_layer  = 6
main.js:1 whisper_model_load: n_mels        = 80
main.js:1 whisper_model_load: f16           = 2
main.js:1 whisper_model_load: type          = 2
main.js:1 whisper_model_load: mem required  =  218.00 MB (+    6.00 MB per decoder)
main.js:1 whisper_model_load: adding 1607 extra tokens
main.js:1 whisper_model_load: model ctx     =  140.60 MB
main.js:1 whisper_model_load: model size    =   54.45 MB
main.js:1 whisper_model_load: ERROR not all tensors loaded from model file - expected 245, got 3
main.js:1 whisper_init_no_state: failed to load model
```

I have also tried to rebuild quantized following ggml repo:

```sh
git clone https://github.com/ggerganov/ggml
cd ggml
mkdir build && cd build
cmake ..
make -j4 whisper-quantize
./bin/whisper-quantize ~/whisper.cpp/models/ggml-base.en.bin ~/whisper.cpp/models/ggml-base.en-q4_0.bin 2
```

but wasm fails on the same error. Any idea how to fix wasm and make it work with quantized models?

Could you also provide update https://whisper.ggerganov.com/ as it seems there are differences to latest master on https://github.com/ggerganov/whisper.cpp/tree/master/examples/whisper.wasm ... i.e `Module.full_default()` has different parapeter count

**Comment by jozefchutka** [comment]
Do I get it right that:

1. demo runs from "4-bit" branch instead of "master"
2. 4-bit will be merged into master in 1.4.0 (following 1.3.0 notes https://github.com/ggerganov/whisper.cpp/releases/tag/v1.3.0 )

**Comment by ggerganov** [comment]
Should be fixed now
